scanners/ssh-audit/parser/parser.js

```diff
@@ -0,0 +1,262 @@
+                description: "SSH encryption algorithms missing",
+                hint: "Add these encryption algorithms"
+            }
+        },
+        addWarning: {
+            kex: {
+                name: "SSH KEX Algorithms must be added",
+                description: "SSH key exchange algorithms missing",
+                hint: "Add these KEX algorithms"
+            },
+            key: {
+                name: "SSH Key Algorithms must be added",
+                description: "SSH key algorithms missing",
+                hint: "Add these key algorithms"
+            },
+            mac: {
+                name: "SSH MAC Algorithms must be added",
+                description: "SSH message authentication code algorithms missing",
+                hint: "Add these MAC algorithms"
+            },
+            enc: {
+                name: "SSH Encryption Algorithms must be added",
+                description: "SSH encryption algorithms missing",
+                hint: "Add these encryption algorithms"
+            }
+        }
+    }
+}
+
+
+/**
+ * Transforms recommendations from the ssh-audit scanner into SSH Policy Violation Findings
+ * @param {String} recommendationSeverityLevel 
+ * @param {{}} value 
+ */
+function transformRecommendationToFinding(recommendationSeverityLevel, value) {
+    // SSH audit has critical and warnings as recommendations. 
+    // These are HIGH and MEDIUM severities, respectively
+    const policyViolationFindings = [];
+    var severity = "low";
+    if (recommendationSeverityLevel == "critical") severity = 'HIGH'
+    if (recommendationSeverityLevel == "warning") severity = 'MEDIUM'
+    const findingTemplate = null;
+    // recommendationAction = del/chg/add
+    Object.entries(value).map(([recommendationAction, algorithms]) => {
+        //algorithmType = kex/ key/ mac, , algorithmNames = {name+note}
+        Object.entries(algorithms).map(([algorithmType, algorithmData]) => {
+            const algorithmNames = []
+            Object.entries(algorithmData).flatMap(([keyNames, content]) => { algorithmNames.push(Object.values(content)) })
+            var action = "";
+            if (recommendationAction == "del" && recommendationSeverityLevel == "critical") action = "delCritical"
+            else if (recommendationAction == "del" && recommendationSeverityLevel == "warning") action = "delWarning"
+            else if (recommendationAction == "chg" && recommendationSeverityLevel == "critical") action = "chgCritical"
+            else if (recommendationAction == "chg" && recommendationSeverityLevel == "warning") action = "chgWarning"
+            else if (recommendationAction == "add" && recommendationSeverityLevel == "critical") action = "addCritical"
+            else if (recommendationAction == "add" && recommendationSeverityLevel == "warning") action = "addWarning"
+            const findingTemplate = templates[action][algorithmType] || null;
```

Comment:
Is there a reason to have this "action" to be build by combining `recommendationAction`and `recommendationSeverityLevel` rather than having both be a new structure in the templates object?

For example if the templates object would look like this

```js
del: {
    critical: {
        kex: {
            name: "Insecure SSH KEX Algorithms",
            ...
        },
        key: ...,
        mac: ...,
        enc: ...,
    },
    warning: {
        kex: {
            name: "Insecure SSH KEX Algorithms",
            ...
        },
        key: ...,
        mac: ...,
        enc: ...,
    },
},
```

you should be able to get the finding template from templates by referencing it by 
`templates[recommendationAction][recommendationSeverityLevel][algorithmType]` without building up this action key.